pandas/tests/indexes/common.py

```diff
@@ -721,3 +736,18 @@ def test_fillna(self):
                 expected[1] = True
                 self.assert_numpy_array_equal(idx._isnan, expected)
                 self.assertTrue(idx.hasnans)
+
+    def test_copy(self):
+        # GH12309
+        for name, index in compat.iteritems(self.indices):
+            if isinstance(index, MultiIndex):
+                continue
+
+            first = index.__class__(index, copy=True, name='mario')
+            second = first.__class__(first, copy=False)
+            self.assertTrue(index.equals(first))
```

Comment:
I think I got the above... not following you here. I am just comparing the values using a public method...